src/journal/journal-file.c

```diff
@@ -1076,23 +1076,24 @@ static int journal_file_append_data(
         o->data.hash = htole64(hash);
 
 #if defined(HAVE_XZ) || defined(HAVE_LZ4)
-        if (f->compress_xz &&
-            size >= COMPRESSION_SIZE_THRESHOLD) {
+        if (JOURNAL_FILE_COMPRESS(f) && size >= COMPRESSION_SIZE_THRESHOLD) {
                 size_t rsize = 0;
 
                 compression = compress_blob(data, size, o->data.payload, &rsize);
 
-                if (compression) {
+                if (compression >= 0) {
```

Comment:
Yes, I actually misread `o->object.flags |= compression;` and thought that it was always setting that to be non-zero (which wouldn't make sense of `compress_blob()` returned 0). This makes sense.